packages/core/src/Stores/client-store.js

```diff
@@ -429,9 +428,11 @@ export default class ClientStore extends BaseStore {
         reaction(
             () => [this.account_settings],
             () => {
-                const lang_from_url = new URLSearchParams(window.location.search).get('lang') || DEFAULT_LANGUAGE;
-                this.setPreferredLanguage(lang_from_url);
-                LocalStore.set(LANGUAGE_KEY, lang_from_url);
+                const language = getRedirectionLanguage(this.account_settings.preferred_language);
```

Comment:
Question: what if this `this.account_settings.preferred_language` is not yet there, isnt it better to have a fallback option just incase ? 🤔

Reply:
inside getRedirectionLanguage there's a fallback language which is `DEFAULT_LANGUAGE`

Reply:
But here what if account_settings is undefined? should we use optional chaining here?

Reply:
will add a optional chaining here 🙏